### PR TITLE
Changed array type

### PR DESCRIPTION
**Pull request by anvaka**
As discussed in https://github.com/toji/gl-matrix/issues/386

### CODE REVIEW

File: src/types.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-interface IndexedCollection {
+interface IndexedCollection extends Iterable<number> {
```
Comment (stefnotch) [inline src/types.d.ts:1]: What's the use case for making it iterable?

Comment (anvaka) [reply src/types.d.ts:1]: Without it passing mat4/vec4 objects to `WebGLRenderingContext.uniformMatrix[234]fv()` fails. See [here](https://github.com/microsoft/TypeScript/blob/deb5bac520a1134e188fc6f01fa4f52b66d2899b/lib/lib.dom.iterable.d.ts#L332-L340).

Comment (stefnotch) [reply src/types.d.ts:1]: Thank you for thoroughly testing everything!

**Comment by anvaka** [comment]
Thank you for the merge! If there are any other issues discovered - I'd be glad to help

**Pull request merged**